### PR TITLE
New render on the server

### PR DESCRIPTION
**Pull request by jamesgorrie**
[koa-views](https://github.com/queckezz/koa-views) is an abstraction for loads of libraries and doesn't allow a lot of control.

We're using `nunjucks`, so this is way simpler and easier to customise.

If we wanted to test out a new view, we definitely could by adding it back in or adding a new render pretty simply.

A couple of ES6 enhancements too - sorry should have been another CR>

### CODE REVIEW

File: server/view/env-utils.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+export function addExtensions(env) {
+  extensions.forEach((Extension, key) => env.addExtension(key, new Extension(env)));
+  return env;
+}
+
+export function getEnvWithExtensions(root) {
+  return addExtensions(getEnv(root));
+}
```
Comment (jamesgorrie) [inline server/view/env-utils.js]: This is abstracted due to the fact that we could use it on the client too.

File: server/extensions/index.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// We could do this automatically with `fs`, but that's unnecessary I/O
+// And doesn't allow us to exclude some.
+module.exports = new Map([]);
```
Comment (jamesgorrie) [inline server/extensions/index.js]: This will be used when we add extensions.

File: server/view/render.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import {getEnvWithExtensions} from './env-utils';
+
+export default function render(root) {
+  return (ctx, next) => {
```
Comment (alfraser86) [inline server/view/render.js]: ctx in this context?

Comment (jamesgorrie) [inline server/view/render.js]: `ctx` is the context used by `Koa`.
It's then passed through all the `use`s, which are generators / async functions.

File: server/view/render.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        env.render(`${relPath}.njk`, templateData, (err, res) => {
+          if (err) {
+            reject(err);
+          } else {
+            ctx.body = res;
+            resolve(res);
+          }
+        });
+
+      })
```
Comment (davidpmccormick) [inline server/view/render.js]: Doesn't the linter want a semi after this?

Comment (jamesgorrie) [inline server/view/render.js]: I'll take a look why this was allowed.

**Pull request merged**